### PR TITLE
admin now gets private info from get /:userId, regular users and lurkers only get public info

### PR DESCRIPTION
**Pull request by jared23m**

### CODE REVIEW

File: BE/db/seed.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+const bcrypt = require("bcrypt");
```
Comment (ljreaux) [inline BE/db/seed.js:1]: Why is this stuff I added showing up in your PR?

File: BE/api/index.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ apiRouter.use(async (req, res, next) => {
   const token = auth?.split(" ")[1];
 
   if (!auth) next();
+  if (token == 'null') next();
```
Comment (ljreaux) [inline BE/api/index.js:16]: I found a better way. Remove this.

File: BE/api/users.js [hunk 3]
```diff
@@ -122,8 +123,16 @@ usersRouter.patch("/me", requireUser, async (req, res, next) => {
 usersRouter.get("/:userId/", async (req, res, next) => {
   const { userId } = req.params;
   try {
-    const user = await getUserPageById(userId);
+    let user;
```
Comment (ljreaux) [inline BE/api/users.js:126]: I think my below code will work a bit better and won't always require you to have an authorization header. 

```js
usersRouter.get("/:userId/", async (req, res, next) => {
  const { userId } = req.params;
  const { user } = req;

  let foundUser;
  try {
    if (user && user.admin) {
      foundUser = await getUserPageById(userId);
    } else {
      foundUser = await getPublicUserPageById(userId);
    }

    res.send(foundUser);
  } catch (err) {
    next(err);
  }
});
```

**Review by ljreaux — APPROVED** [review]
I looked over it. the line in api/index.js is causing the issue. Remove it. Also are you aware that while this does exclude email from the initial user, the email is still included in the attached comments?

File: BE/db/seed.js (hunk 1 above)
Comment (jared23m) [reply BE/db/seed.js:1]: haha i added it. I needed it for testing. You can't stop me boiiiiiiiiiii

**Review by ljreaux — APPROVED** [review]

**Review by BarryFullStack — APPROVED** [review]

**Review by amead99 — APPROVED** [review]

**Pull request merged**